src/client/remoteinputhandler.cpp

```diff
@@ -149,86 +146,61 @@ void RemoteInputHandler::step(float dtime) {
 };
 
 void RemoteInputHandler::step_post_render() {
-  if (m_should_send_observation) {
-    m_should_send_observation = false;
-
-    // send current observation
-    irr::video::IVideoDriver *driver = m_rendering_engine->get_video_driver();
-    irr::video::IImage *image = driver->createScreenShot(video::ECF_R8G8B8);
-
-    // parse score from hud
-    // during game startup, the hud is not yet initialized, so there'll be no
-    // score for the first 1-2 steps
-    float score{};
-    std::map<std::string, float> aux{};
-    for (u32 i = 0; i < m_player->maxHudId(); ++i) {
-      auto hud_element = m_player->getHud(i);
-      std::string_view elem_text = hud_element->text;
-      // find the index of the first :
-      const auto colon_pos = elem_text.find(':');
-      if (colon_pos == std::string_view::npos) {
-        continue;
-      }
-      elem_text.remove_prefix(colon_pos + 1); // +1 for space
-      // I'd rather use std::from_chars, but it's not available in libc++ yet.
-      std::stringstream ss{std::string(elem_text)};
-      if (hud_element->name == "score") {
-        ss >> score;
-      } else {
-        float value{};
-        ss >> value;
-        aux[hud_element->name] = value;
-      }
-    }
-
-    // The non-referential type here ensures that we deep-copy the strings while we hold the lock.
-    std::unordered_map<std::string, std::string> player_meta;
-    {
-      std::lock_guard<ordered_mutex> server_lock = std::lock_guard<ordered_mutex>(gServer->getEnvMutex());
-      auto remote_player = gServer->getEnv().getPlayer(m_player->getName());
-      assert(remote_player != nullptr);
-      player_meta = remote_player->getPlayerSAO()->getMeta().getStrings();
-    }
+  if (!m_should_send_observation) {
+    return;
+  }
 
-    {
-      auto builder_buffer = new capnp::MallocMessageBuilder();
+  m_should_send_observation = false;
 
-      auto obs_builder = builder_buffer->initRoot<Observation>();
-      obs_builder.setReward(score);
+  auto builder_buffer = new capnp::MallocMessageBuilder();
```

Comment:
```suggestion
  auto message_builder = new capnp::MallocMessageBuilder();
```

Reply:
I think the capnp name isn't so great. It should have been called something-arena or -buffer. I wanted to emphasize it's role.